pkg/tempo/config.go

```diff
@@ -167,10 +169,61 @@ func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 		retryConfig["max_elapsed_time"] = "60s"
 	}
 
-	otelMapStructure["exporters"] = map[string]interface{}{
-		"otlp": otlpExporter,
+	return otlpExporter, nil
+}
+
+// exporters builds one or multiple exporters from a remote_write block.
+// It also supports building an exporter from push_config.
+func (c *InstanceConfig) exporters() (map[string]interface{}, error) {
+	if len(c.RemoteWrite) == 0 {
+		otlpExporter, err := c.exporter(c.PushConfig)
+		return map[string]interface{}{
+			"otlp": otlpExporter,
+		}, err
+	}
+
+	exporters := map[string]interface{}{}
+	for i, remoteWriteConfig := range c.RemoteWrite {
+		exporter, err := c.exporter(remoteWriteConfig)
+		if err != nil {
+			return nil, err
+		}
+		exporterName := fmt.Sprintf("otlp/%d", i)
+		exporters[exporterName] = exporter
+	}
+	return exporters, nil
+}
+
+func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
+	otelMapStructure := map[string]interface{}{}
+
+	if len(c.Receivers) == 0 {
+		return nil, errors.New("must have at least one configured receiver")
+	}
+
+	if len(c.RemoteWrite) == 0 && len(c.PushConfig.Endpoint) == 0 {
+		return nil, errors.New("must have a configured at least one remote_write or push_config.endpoint")
+	}
+
+	if len(c.RemoteWrite) != 0 && len(c.PushConfig.Endpoint) != 0 {
```

Comment:
instead of testing this we could just concatenate PushConfig with the new RemoteWrite slice?  we could remove this error checking.

maybe just have a 
```
if len(c.PushConfig.Endpoint) > 0 {
  // print friendly deprecation warning here.
}
```

Reply:
That's a great idea. I hadn't thought of that.

However, I'm not entirely convinced.  I think it could result in unexpected or undesired behaviour when defining exporters with both `push_config` and `remote_write`. I think it's convenient to check that only one of the methods is being used.